decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
+end
```

Comment:
- [ OPTIONAL ] Great job working on the functionality so far but I think you forget to add a condition here to first makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word. so you should check if it is longer than 10 trim it is not just keep it I assigned it as optional because the privous reviewer didn't mention it but it still important to add! so kindly do it 💪🏻⭐🔥✨

## If you want to trim the word to have a maximum 10 char so you can slice from (0, 9) since you started from 0 but if you want to start from 1 you can add it (1, 10) 🔥🔥💪🏻

![image](https://github.com/calebchris000/School_Library/assets/58553711/37ad5b5a-764c-4c25-8863-f4997c37e4f7)